src/services/stringCompletions.ts

```diff
@@ -938,10 +951,41 @@ function getCompletionEntriesForNonRelativeModules(
             }
         }
         if (!foundGlobal) {
+            let seenPackageScope = false;
             let ancestorLookup: (directory: string) => void | undefined = ancestor => {
                 const nodeModules = combinePaths(ancestor, "node_modules");
                 if (tryDirectoryExists(host, nodeModules)) {
-                    getCompletionEntriesForDirectoryFragment(fragment, nodeModules, extensionOptions, host, /*moduleSpecifierIsRelative*/ false, /*exclude*/ undefined, result);
+                    getCompletionEntriesForDirectoryFragment(fragment, nodeModules, extensionOptions, compilerOptions, host, /*moduleSpecifierIsRelative*/ false, /*exclude*/ undefined, result);
+                }
+                if (!seenPackageScope) {
+                    const packageFile = combinePaths(ancestor, "package.json");
+                    if (seenPackageScope = tryFileExists(host, packageFile)) {

```

Comment:
TODO: this has to be heavily deduplicated with the existing branch that handles `exports`